bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/activities/MutableActivityManager.java

```diff
@@ -133,6 +133,23 @@ public MutableActivityManager(ITriggerPointAdvisor triggerPointAdvisor, IActivit
 		readRegistry(true);
 	}
 
+	@Override
+	public void dispose() {
+		super.dispose();
+		clearExpressions();
+		this.activityRegistry.removeActivityRegistryListener(activityRegistryListener);
+		this.activitiesById = null;
```

Comment:
Thanks for pointing that out. From my point of view, scenarios where an activity manager is (unintendedly) used after disposing it are unlikely. The API is internal and we know about its current usages, which is basically only during shutdown at which no further access to it is performed.

I see two lines of argumentation: 
1. You could argue that throwing NPEs in case of inappropriate API usage is good because you get fast feedback that you are doing unintended things. 
2. You could argue that the risk of NPEs is unnecessary as there is no risk of memory leaks and no performance benfit when nulling the values instead of clearing them. And the occurence of NPEs on inappropriate API usage may not occur during development but may be restricted to specific usage scenarios because of a race condition.

I think that both argumentations are valid but I tend to follow your proposal of being lenient (so going with argument 2). I adapted the PR with ada4186.